### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 25

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#265] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152317155-8280652f-99db-49e3-9f27-8c0458481c84.png)

**Comment by matypist** [comment]
> 1. F perché f non può essere iniettiva
> 2. V  
> 3. V

_Postato originariamente da @i0nut02 su https://github.com/sapienzastudentsnetwork/mmi2122/issues/171_

**Comment by matypist** [comment]
Io all'esame ho risposto:

1. V
2. F
3. V

Verificherò meglio le mie risposte rivedendo le dispense

**Comment by LuginiAndrea** [comment]
La prima è falsa.
$f$ non può essere iniettiva in quanto # $B$ < # $A$

La seconda è vera. In quanto $\exists x_1, x_2 \in A$ t.c. $x_1 \neq x_2, f\left(x_1\right) = f\left(x_2\right) \Longrightarrow b \in B$ t.c. $b = f\left(x_1\right) = f\left(x_2\right)$ provando quindi che $g$ non può essere invertibile in quanto $g\left(b\right)$ dovrebbe essere al contempo uguale a $x_1$ ed a $x_2$

La terza è vera: Riferirsi all'esercizio 24

**Comment by TheNewHEROBRINEX** [comment]
Secondo me la prima è vera, in accordo con la tavola di verità dell'operatore se ... allora ...
In quanto se l'antecedente è falso allora la proposizione è vera indipendentemente dal valore del conseguente.

**Comment by Elia-Belli** [comment]
- La **(1)** può essere ambigua, per me è vera e ne do una mia intepretazione: possiamo riscrivere la frase sostituendo al costrutto "se allora" un'espressione equivalente, ovvero "L'iniettività di f e g è condizione sufficiente per l'iniettività di (g o f)", per la **Caratterizzazione dell'Iniettività** quest'affermazione è vera. Ovviamente f non è iniettiva e quindi dire (g o f) iniettiva è sbagliato, ma l'affermazione che ci viene presentata è vera (**nella consegna è chiesto di determinare se le affermazioni sono corrette**). 
- La **(2)** è vera: a causa della cardinalità, f è costretta ad associare almeno 2 elementi distinti in A allo stesso elemento in B, di conseguenza g non può restituire entrambe le preimmagini ma solo una (altrimenti non sarebbe una funzione), quindi è impossibile che (g o f) sia l'identità su A.
- La **(3)** è vera: ovvia, sia che g agisca su B che sull'immagine di f contenuta in B, tutti gli elementi vengono sempre trasformati in 1.
